src/OpenCensus.Collector.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -43,7 +46,9 @@ public override void OnStartActivity(Activity activity, object payload)
 
             var request = context.Request;
 
-            this.Tracer.SpanBuilder("HttpIn").SetSampler(this.Sampler).StartScopedSpan();
+            var ctx = this.propagationComponent.TextFormat.Extract<HttpRequest>(request, (r, name) => r.Headers["name"].ToString());
```

Comment:
r.Headers["name"] -> r.Headers[name]

and please do TryGet and assume there could be multiple of them or none at all

Reply:
fixed. thank you